src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -2538,6 +2537,23 @@ public static void SortDescriptorArray(IList infos)
             ArrayList.Adapter(infos).Sort(MemberDescriptorComparer.Instance);
         }
 
+        /// <summary>
+        /// Wraps ComWrappers.TryGetComInstance and returns false when not supported
+        /// </summary>
+        private static unsafe bool TryGetComInstance(object instance, out IntPtr unknown)
+        {
+            unknown = IntPtr.Zero;
+            try
+            {
+                return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
```

Comment:
Would it be better to change `ComWrappers.TryGetComInstance` to return false when ComWrappers are not supported instead?

It makes sense to throw PNSE exceptions from factories. It does not make sense to thrown PNSE from APIs that check whether the instance was created by given factory, like this API. If the factory throws PNSE, we are guaranteed that the instance was not created by the given factory and thus it is safe to just return false instead of making all callers to catch PNSE.
 

cc @AaronRobinsonMSFT

Reply:
For example, `Type.IsCOMObject` return false. It does not throw PNSE when COM is not supported.

Reply:
#90553 is the alternative fix that I have described.

Reply:
So this proposal was something I was considering, but I was struggling with when we decide to throw PNSE or when we return `false`. It is entirely possible this API is niche enough that returning `false` is the right thing, in fact, I would prefer it. However, I would like to understand when we return `false` from an API rather than throwing PNSE.

Reply:
If the API is a constructor, an instance factory or a static setter, throw PNSE.

If the API is a getter, return the default value that would have been returned when the feature is not active.

Here is a recent example of a similar fix where we have changed an API to return the default value instead of throwing PNSE: https://github.com/dotnet/runtime/pull/87278/

Reply:
@jkotas Okay, approved. I've signed off on that approach.

@ericstj Any concerns with the approach proposed by @jkotas?

Reply:
This change was meant to just fix the regression, not to reinvent the original fix. The same suggestion came up with the original fix but was not done at that time.  If folks want to make that change to ComWrappers now I will go ahead and close this PR.